src/main/java/teammates/common/datatransfer/FeedbackQuestionAttributes.java

```diff
@@ -510,14 +509,10 @@ public void sanitizeForSaving() {
     
     public boolean isJsonValid(String jsonString) {
         try {
-            // Checks whether JSONArray is valid
-            new JSONObject(jsonString);
-        } catch (JSONException e) {
-            try {
-                new JSONArray(jsonString);
-            } catch (JSONException ex) {
-                return false;
-            }
+            JsonParser parser = new JsonParser();
+            parser.parse(jsonString);
```

Comment:
@tarun14110, this can't be right. Gson's JsonParser is very lenient so even `new JsonParser().parse("123");` does not throw an exception.